src/Common/MoistThermodynamics/states.jl

```diff
@@ -191,16 +154,31 @@ struct PhaseNonEquil{FT} <: ThermodynamicState{FT}
 end
 
 """
-    LiquidIcePotTempSHumNonEquil_no_ρ(θ_liq_ice, q_tot, p)
+    LiquidIcePotTempSHumNonEquil(θ_liq_ice, q_pt, ρ)
+
+Constructs a [`PhaseNonEquil`](@ref) thermodynamic state from:
+
+ - `θ_liq_ice` - liquid-ice potential temperature
+ - `q_pt` - phase partition
+ - `ρ` - density
+"""
+function LiquidIcePotTempSHumNonEquil(θ_liq_ice::FT, q_pt::PhasePartition{FT}, ρ::FT) where {FT<:Real}
+    T = air_temperature_from_liquid_ice_pottemp(θ_liq_ice, ρ, q_pt)
+    e_int = internal_energy(T, q_pt)
+    PhaseNonEquil(e_int, q_pt, ρ)
+end
+
+"""
+    LiquidIcePotTempSHumNonEquil_given_pressure(θ_liq_ice, q_pt, p)
 
 Constructs a [`PhaseNonEquil`](@ref) thermodynamic state from:
 
  - `θ_liq_ice` - liquid-ice potential temperature
  - `q_pt` - phase partition
  - `p` - pressure
 """
-function LiquidIcePotTempSHumNonEquil_no_ρ(θ_liq_ice::FT, q_pt::PhasePartition{FT}, p::FT) where {FT<:Real}
-    T = θ_liq_ice * exner(p, q_pt)
+function LiquidIcePotTempSHumNonEquil_given_pressure(θ_liq_ice::FT, q_pt::PhasePartition{FT}, p::FT) where {FT<:Real}
+    T = air_temperature_from_liquid_ice_pottemp_given_pressure(θ_liq_ice, p, q_pt)
     ρ = air_density(T, p, q_pt)
     e_int = internal_energy(T, q_pt)
     PhaseNonEquil(e_int, q_pt, ρ)
```

Comment:
Can you add the following as well:

`"""                                                                                                                                                                                                                                    
    LiquidIcePotTempSHumEquil_given_pressure(θ_liq_ice, q_pt, p)                                                                                                                                                                       
                                                                                                                                                                                                                                       
Constructs a [`PhaseEquil`](@ref) thermodynamic state from:                                                                                                                                                                            
                                                                                                                                                                                                                                       
 - `θ_liq_ice` - liquid-ice potential temperature                                                                                                                                                                                      
 - `q_pt` - phase partition                                                                                                                                                                                                            
 - `p` - pressure                                                                                                                                                                                                                      
"""
function LiquidIcePotTempSHumEquil_given_pressure(θ_liq_ice::FT, q_pt::PhasePartition{FT}, p::FT) where {FT<:Real}
    T = air_temperature_from_liquid_ice_pottemp_given_pressure(θ_liq_ice, p, q_pt)
    ρ = air_density(T, p, q_pt)
    e_int = internal_energy(T, q_pt)
    PhaseEquil(e_int, q_pt, ρ)
end

`

Reply:
I'll try to do this in a later PR (I tried to add this recently, and I had to add a _lot_ more functions).